### PR TITLE
feat(cart): add total_discount to DaffCartItem

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: #719 

## What is the new behavior?
- Adds `total_discount` to `DaffCartItem`
- Transforms `total_discount` in the Magento cart item transformer

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/cart/src/drivers/magento/transforms/outputs/cart-item.service.ts [hunk 1]
```diff
@@ -29,9 +29,10 @@ export class DaffMagentoCartItemTransformer {
       product_id: String(cartItem.product.id),
 			image: {
 				id: cartItem.product.image.label,
-				url: cartItem.product.image.url, 
+				url: cartItem.product.image.url,
 				label: cartItem.product.image.label
-			},
+      },
+      discounted_row_total: cartItem.prices.row_total.value - cartItem.prices.total_item_discount.value,
```
Comment (lderrickable) [inline libs/cart/src/drivers/magento/transforms/outputs/cart-item.service.ts]: I'm pretty sure the total_item_discount is the total of discounts applied to an item without consideration of the quantity. The row_total does consider quantity.  We should probably just put a total_discounts on the DaffCartItem and return the discounted row total in a selector. That way, as we add more platforms, we don't need to keep doing this computation over and over (presuming other platforms also don't have a ready-made discounted_row_total field).

Comment (griest024) [reply libs/cart/src/drivers/magento/transforms/outputs/cart-item.service.ts]: Doing that computation in a selector seems better. Changed.

File: libs/cart/testing/src/factories/cart-item.factory.ts [hunk 2]
```diff
@@ -14,14 +14,15 @@ export class MockCartItem implements DaffCartItem {
   qty = faker.random.number({min:1, max:100});
   price = faker.random.number(1500);
   row_total = this.qty * this.price
+  total_discount = faker.random.number(1000);
```
Comment (lderrickable) [inline libs/cart/testing/src/factories/cart-item.factory.ts]: I think we should guarantee that this will be less than the price field

Comment (griest024) [reply libs/cart/testing/src/factories/cart-item.factory.ts]: Changed

**Pull request merged**